### PR TITLE
Docs: git pull is needed after syncing git modules

### PR DESCRIPTION
**Pull request by Apollon77**
I followed the instructions to update the repo but still wondered why it dit not really do anything. Now I stumbled the secod time over the fact that tthe "clone" command in docs give you a full working repo, but the "synching modules" does not update them ... so git pull is needed.

I think it nmight be good to add that here in order to support users.

### CODE REVIEW

File: docs/guides/BUILDING.md [hunk 1]
```diff
@@ -40,6 +40,12 @@ synchronize submodules:
 git submodule update --init
 ```
 
+After synchronizing the modules you need to pull the latest changes:
```
Comment (bzbarsky-apple) [inline docs/guides/BUILDING.md:43]: Pull the latest changes of what?  `submodule update --init` should be updating all the modules to the relevant SHAs...

If you are trying to document how to fully update a repo to a new SHA, that would be a `git pull` _followed_ by a submodule update?

Comment (Apollon77) [reply docs/guides/BUILDING.md:43]: Ok if the git pull needs to be done before the sync, also ok (but the docs also do notstate this).

I can try that oncve the next PR got merged here on master and see if this works for me. I will then update here

Comment (bzbarsky-apple) [reply docs/guides/BUILDING.md:43]: You're right that the docs don't say that, it being basic git.  Fixing them to say that would be a good idea.  I just don't want them giving the idea that you do pull after submodule update.  ;)

Comment (Apollon77) [reply docs/guides/BUILDING.md:43]: All good ... I'm also just a "git user" and in fact it worked for me that way, biut as said I will try your approach and update the PR. Will set to draft till then

Comment (Apollon77) [reply docs/guides/BUILDING.md]: Verified and Updated

**Comment by Apollon77** [comment]
@bzbarsky-apple ready for re-review :-)

**Review by bzbarsky-apple — APPROVED** [review]

File: docs/guides/BUILDING.md [hunk 2]
```diff
@@ -31,12 +31,13 @@ To check out the Matter repository, run the following command:
...
-## Synchronizing submodules
+## Updating Matter code
 
-If you already have the Matter code checked out, run the following command to
+If you already have the Matter code checked out, run the following commands to update the repository and 
 synchronize submodules:
 
 ```
+git pull
 git submodule update --init
```
Comment (andy31415) [inline docs/guides/BUILDING.md:41]: I wonder if `git submodule update -f --init --recursive` is better here. Sometimes I also needed a sync.

**Review by andy31415 — APPROVED** [review]

File: docs/guides/BUILDING.md (hunk 2 above)
Comment (Apollon77) [reply docs/guides/BUILDING.md:41]: True, I also have the "feeling" that it might be better, but I am not deep enough in git. Should I add it?

Comment (andy31415) [reply docs/guides/BUILDING.md:41]: I think you applied the change that you needed, if you never needed my call it is fine. If you ever need the recursive or -f please create a PR to add it here :)

**Comment by andy31415** [comment]
@Apollon77 note the restyle error - please apply that patch and then this should auto-merge.

**Comment by andy31415** [comment]
> @Apollon77 note the restyle error - please apply that patch and then this should auto-merge.

https://restyled.io/gh/project-chip/repos/connectedhomeip/jobs/3567122 has the instructions (click on the red mark near the "restyled" error)

**Comment by Apollon77** [comment]
@andy31415 I just did. But honestly ... this restyled ... impossible to be more cryptic ... it should also just tell whats up and whats wrong/to change ;-)

File: docs/guides/BUILDING.md (hunk 2 above)
Comment (bzbarsky-apple) [reply docs/guides/BUILDING.md:41]: @andy31415 Fwiw, doing `--recursive` stuff on our submodules can get some of them in bad states, from what I have seen.  So it's actually sort of a bad idea.  :(

Comment (Apollon77) [reply docs/guides/BUILDING.md:41]: Good to know :-) thank you!

**Pull request merged**